### DISCUSSION TITLE
https://mjasion.pl/posts/kubernetes/how-to-debug-istio-upstream-reset/?utm_source=jvm-bloggers.com&utm_medium=link&utm_campaign=jvm-bloggers#double-transfer-encoding-header---what-does-it-mean

### DISCUSSION THREAD

**Discussion by giscus** [mjasion/mjasion#151] [General]
# https://mjasion.pl/posts/kubernetes/how-to-debug-istio-upstream-reset/?utm_source=jvm-bloggers.com&utm_medium=link&utm_campaign=jvm-bloggers#double-transfer-encoding-header---what-does-it-mean

Istio can reset processing the request. This blog post shows how to analyze the issue if logs does not help

https://mjasion.pl/posts/kubernetes/how-to-debug-istio-upstream-reset/?utm_source=jvm-bloggers.com&utm_medium=link&utm_campaign=jvm-bloggers

**Comment by saulgill** [comment]
How did you fix it? Is Istio working for you now? I believe I have exactly the same issue.

**Reply by mjasion** [reply]
Hi @saulgill
If you are facing the same issue with your app, what you need is to remove adding a duplicated header(e.g. your app adds a header in uppercase). If this is uppercase, then change the header to lowercase.
Istio is strict here, that this is an application issue because it breaks the HTTP protocol. 

Please take a look at this [Istio issue](https://github.com/istio/istio/issues/24753#issuecomment-656380098) and this answer at [envoy pull request](https://github.com/envoyproxy/envoy/pull/11916#issuecomment-655490517) which explains why making workarounds are not accepted.
